### ISSUE TITLE
[API] Séparation backend/frontend

### ISSUE THREAD

**Issue by Hydrog3n** [mondediefr/mondedie-chat#12]
Il serait bien de bien séparer le backend du front end afin de pouvoir faire différent clients et d'utiliser une API. Quand pensez vous ?

**Comment by hardware** [comment]
Oui bonne idée, tu prévois de faire un client iOS avec Swift ?

**Comment by Hydrog3n** [comment]
Je pensais plus à une encapusulation avec du cordova comme ça pas besoin de refaire tout les accès socket. Il faudra surrement mettre ne place l'api avec.

**Comment by soyuka** [comment]
Je trouve que c'est déjà bien séparé, qu'entends-tu par là ?

(en gros je viens flood dans une issue pour dire :clap: pck vous avez fait du super taf :D)

**Comment by Hydrog3n** [comment]
Je veux dire par là l'on puisse les lancer séparément. Sur des instances différentes.

**Comment by soyuka** [comment]
ah bah c'est presque le cas là, basiquement faut juste bouger la route [/ linked](https://github.com/mondediefr/mondedie-chat/blob/master/routes/index.js#L16) qui render la page \o/.

**Comment by Hydrog3n** [comment]
Je pense pas comprendre là. 

Pour moi : le index.js c'est pour le serveur la partie front devrait être lancé indépendamment comme ça. On pourrait l'encapsuler très facilement dans nwjs ou autre.

**Comment by soyuka** [comment]
Bah oui y'a juste le render qui doit sortir et c'est bon :) tout le reste
c'est du pur server

Le jeu. 26 nov. 2015 à 14:33, Loïc Vaillé notifications@github.com a
écrit :

**Comment by Hydrog3n** [comment]
Exact ! ^^'

**Comment by soyuka** [comment]
J'avais commencé à fork votre projet, j'aime beaucoup l'interface et je voulais prendre la partie front, l'injécter dans electron et à la place de l'api placer un webrtc-swarm\* (https://github.com/mafintosh/webrtc-swarm utilisé par friends (https://github.com/moose-team) et une surcouche PGP pour crypter les messages le tout en relation avec keybase.
- https://github.com/peermusic/secure-webrtc-swarm => keybase ftw

**Comment by Hydrog3n** [comment]
ok je voulais placer le front dans du cordova pour le mobile pour ma part.

**Comment by soyuka** [comment]
:+1: 

https://github.com/mondediefr/mondedie-chat/blob/master/routes/index.js#L16

Ce que je ferais pour jarter cette route :
1. Settings via un endpoint de l'api `GET /settings`
2. index.html est chargé via nginx, pas besoin de `GET /` sur l'api
3. Lorsque ca charge, ca `GET /settings` et ca init le reste après

**Comment by Hydrog3n** [comment]
Yes, On va devoir créer un xhr pour reccup ça. Ou tu as une autre idée ?

**Comment by soyuka** [comment]
Hmm oui, d'ailleurs il faudrait aussi login en SPA. Cas où l'api de login peut déjà renvoyer les settings.
